src/AsmResolver.DotNet/TypeDefinition.cs

```diff
@@ -42,6 +42,81 @@ public class TypeDefinition :
         private IList<InterfaceImplementation>? _interfaces;
         private IList<MethodImplementation>? _methodImplementations;
 
+        /// <summary>
+        /// Initialize New Delegate.
+        /// </summary>
+        /// <param name="ns">Delegate Namespace.</param>
+        /// <param name="name">Delegate Name.</param>
+        /// <param name="module">Delegate Module.</param>
+        /// <param name="returnType">Invoke Return Type.</param>
+        /// <param name="parameterTypes">Invoke Parameters Types.</param>
+        /// <param name="attributes">Accessibility Attributes.</param>
+        /// <returns>Fully Implemented Delegate.</returns>
+        public static TypeDefinition CreateDelegate(string ns, string name, ModuleDefinition module, TypeSignature returnType, IEnumerable<TypeSignature> parameterTypes, TypeAttributes attributes = TypeAttributes.NotPublic) =>
+            CreateDelegate(ns, name, module, MethodSignature.CreateInstance(returnType, parameterTypes), attributes);
+
+        /// <summary>
+        /// Initialize New Delegate.
+        /// </summary>
+        /// <param name="ns">Delegate Namespace.</param>
+        /// <param name="name">Delegate Name.</param>
+        /// <param name="module">Delegate Module.</param>
+        /// <param name="invokeSignature">Delegate.Invoke Signature.</param>
+        /// <param name="attributes">Accessibility Attributes.</param>
+        /// <returns>Fully Implemented Delegate.</returns>
+        public static TypeDefinition CreateDelegate(string ns, string name, ModuleDefinition module, MethodSignature invokeSignature, TypeAttributes attributes = TypeAttributes.NotPublic)
+        {
+
+            var importer = new ReferenceImporter(module);
+
+            var asyncResult = importer.ImportType(new TypeReference(module.CorLibTypeFactory.CorLibScope, nameof(System), nameof(IAsyncResult))).ToTypeSignature();
```

Comment:
We probably want to avoid using `ToTypeSignature()` here for performance reasons, since it calls the assembly resolver for determining the value of `IsValueType`, but we always know the value of `IsValueType`.